### PR TITLE
first version of hw

### PR DESCRIPTION
**Pull request by oleksiistepaniak**

### CODE REVIEW

**Review by KyslyiViacheslav — APPROVED** [review]
Good job

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```
Comment (sarakhmen) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:123]: do not forget also to assign corresponding drivers to each car here

File: src/main/java/mate/jdbc/dao/CarDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars with driver id "
+            + driverId, e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
```
Comment (sarakhmen) [inline src/main/java/mate/jdbc/dao/CarDaoImpl.java:144]: I'd also recommend checking if drivers are empty here

**Review by sarakhmen — CHANGES_REQUESTED** [review]
Let's improve your solution :) Also do not forget to put your implementations into 'impl' package

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+        for (Car car: cars) {
```
Comment (nacenik) [inline src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java:84]:
```suggestion
        for (Car car : cars) {
```

File: src/main/resources/init_db.sql [hunk 4]
```diff
@@ -18,3 +18,29 @@ CREATE TABLE `drivers` (
...
                                   UNIQUE INDEX `license_number_UNIQUE` (`license_number` ASC) VISIBLE);
+
+DROP TABLE IF EXISTS `cars`;
+CREATE TABLE `cars` (
+                                  `id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+                                  `manufacturer_id` BIGINT,
+                                  `model` VARCHAR(225) NOT NULL,
+                                  `is_deleted` TINYINT NOT NULL DEFAULT 0,
+                                  PRIMARY KEY (`id`),
+
```
Comment (nacenik) [inline src/main/resources/init_db.sql:29]: Redundant empty space

File: src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id, manufacturer_id, model, name, country"
+                + " FROM cars c"
+                + " JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.is_deleted = FALSE";
```
Comment (nacenik) [inline src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java:49]: Add semicolons in all queries.

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution! Let’s improve it 😉

**Review by nacenik — APPROVED** [review]
LGTM! 👍

**Pull request closed without merging**